File: Python/So_nguyen_to.py
from math import sqrt

def soNguyenTo(n):
    count = int(0)
    canBacHai = int(sqrt(n))
    for i in range(2,canBacHai+1,1):
        if(n%i==0):
            count+=1
    if(count==0):
        print("YES")
    else:
        print("NO")

n = int(input())
if(n==1):
    print("NO")
elif(n==2 or n==3):
    print("YES")
else:
    soNguyenTo(n)

File: C_C++/So_nguyen_to.cpp
#include <iostream>
#include <cmath>
using namespace std;

void soNguyenTo(unsigned long long n)
{
	unsigned long long i;
	int count = 0;
	for(i=2;i<=sqrt(n);i++)
	{
		if(n%i==0)
			count++;
	}
	if(count==0)
		cout << "YES";
	else
		cout << "NO";
}

int main()
{
	unsigned long long n;
	cin >> n;
	if(n==1)
		cout << "NO";
		else if(n==2 || n==3)
			cout << "YES";
	else
		soNguyenTo(n);
	return 0;
}